src/test/scala/mesosphere/marathon/api/akkahttp/v2/GroupsControllerTest.scala

```diff
@@ -33,6 +31,22 @@ class GroupsControllerTest extends UnitTest with ScalatestRouteTest with Inside
   implicit val identity: Identity = new Identity {}
 
   "Group detail" should {
+
+    // Entity not found test cases
+    {
+      val f = new FixtureWithRealGroupManager(authorized = false)
+      behave like unknownEntity(forRoute = f.groupsController.route, withRequest = Delete("/groupname"), withMessage = "Group '/groupname' does not exist")
+
+      val infoService = mock[GroupInfoService]
+      infoService.selectGroup(any, any, any, any) returns Future.successful(None)
+      infoService.selectGroupVersion(any, any, any, any) returns Future.successful(None)
+      val g = new Fixture(infoService = infoService)
+
+      behave like unknownEntity(forRoute = g.groupsController.route, withRequest = Get("/unknown-group"), withMessage = "Group '/unknown-group' does not exist")
+      behave like unknownEntity(forRoute = g.groupsController.route, withRequest = Get("/groupname/versions/2017-10-30T16:08:53.852Z"), withMessage = "Group '/groupname' does not exist in version 2017-10-30T16:08:53.852Z")
```

Comment:
Where does the magical timestamp of `2017-10-30T16:08:53.852Z` come from? `SettableClock` already has `2015-04-09T12:30:00` as a magical constant.

Reply:
`2017-10-30T16:08:53.852Z` is an unknown versions. That's what the test is testing 🙂.